dotcms-integration/src/test/java/com/dotcms/timemachine/business/TimeMachineAPITest.java

```diff
@@ -0,0 +1,131 @@
+package com.dotcms.timemachine.business;
+
+import com.dotcms.util.IntegrationTestInitService;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.FileUtil;
+import junit.framework.TestCase;
+import org.jetbrains.annotations.NotNull;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class TimeMachineAPITest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to test: {@link TimeMachineAPIImpl#removeOldTimeMachineBackupsFiles()}
+     * When: You have 10 folder with a lastModified older than the PRUNE_TIMEMACHINE_OLDER_THAN_DAYS default value (90)
+     * and you have 1o folder with a lastModified younger than he PRUNE_TIMEMACHINE_OLDER_THAN_DAYS default value (90)
+     * Should: Remove the folders with a older value
+     */
+    @Test
+    public void removeOldTimeMachineBackupsFiles() throws IOException {
+
+        final File tmTestingFolder = FileUtil.createTemporaryDirectory("tm_testing");
+        final String timemachinePathPreviousValue = Config.getStringProperty("TIMEMACHINE_PATH", null);
+
+        try {
+            Config.setProperty("TIMEMACHINE_PATH", tmTestingFolder.toPath().toAbsolutePath().toString());
+
+            final List<File> expireFolders = createFiles(tmTestingFolder, "timeMachineBundle_expired_",
+                    91, 100);
+            final List<File> notExpireFolders = createFiles(tmTestingFolder, "timeMachineBundle_not_expired_",
+                    45, 89);
+
+            final List<File> removedFiles = APILocator.getTimeMachineAPI().removeOldTimeMachineBackupsFiles();
+
+            assertEquals(expireFolders.size(), removedFiles.size());
+            assertTrue(removeAll(expireFolders, removedFiles).isEmpty());
+
+            final File[] files = tmTestingFolder.listFiles();
+            assertEquals(notExpireFolders.size(), files.length);
+            assertTrue(removeAll(Arrays.asList(), notExpireFolders).isEmpty());
```

Comment:
this looks incomplete: "Arrays.asList()"

Reply:
done https://github.com/dotCMS/core/pull/28530/commits/6e6757d800ae36575a4fd3854c1a6e164d8bd327#diff-f27467cb9919462f8c7c5a137c3c9b9bf21be349047c0d1e1ba9315773e53298R55